src/shared/clients/ec2Client.ts

```diff
@@ -11,35 +11,48 @@ import {
     Reservation,
     DescribeInstanceStatusRequest,
     InstanceStateName,
+    Tag,
 } from '@aws-sdk/client-ec2'
 import { AsyncCollection } from '../utilities/asyncCollection'
 import { pageableToCollection } from '../utilities/collectionUtils'
 import { IamInstanceProfile } from 'aws-sdk/clients/ec2'
 
+export interface Ec2Instance {
```

Comment:
would it make senes to `extends` the sdk types, to avoid having to re-map properties. Example from the codecatalyst client: https://github.com/aws/aws-toolkit-vscode/blob/38e13ea204d375ea9b1a4ee1e62075257760c432/src/shared/clients/codecatalystClient.ts#L83

Reply:
Yeah, that makes sense. I changed it to extend the sdk type, but add the name field. The name is traditionally buried in the tags so this makes it easier to work with.